catalyst-gateway/bin/src/service/api/documents/post_document_index_query/mod.rs

```diff
@@ -35,6 +36,11 @@ pub(crate) enum Responses {
     /// The Index of documents which match the query filter.
     #[oai(status = 200)]
     Ok(Json<DocumentIndexListDocumented>),
+    /// ## OK
+    ///
+    /// The Index of documents which match the query filter.
+    #[oai(status = 200)]
+    OkV2(Json<v2::response::DocumentIndexListDocumentedV2>),
```

Comment:
This cannot been used for the `v1` endpoint at all, otherwise it does not make sense to make a separate `v2` endpoint.
Move all the code related to the `v1` endpoint under its own mod